api/src/main/java/org/pmiops/workbench/monitoring/MonitoringSpringConfiguration.java

```diff
@@ -25,4 +42,38 @@ public StatsRecorder getStatsRecorder() {
   public Tagger getTagger() {
     return Tags.getTagger();
   }
+
+  @Bean(name = APP_ENGINE_NODE_ID)
+  public String getAppEngineNodeId(ModulesService modulesService) {
+    try {
+      return modulesService.getCurrentInstanceId();
+    } catch (ModulesException e) {
+      return makeRandomNodeId();
+    }
+  }
+
+  @Bean
+  public MonitoredResource getMonitoredResource(
+      @Qualifier(CacheSpringConfiguration.WORKBENCH_CONFIG_SINGLETON)
+          Provider<WorkbenchConfig> workbenchConfigProvider,
```

Comment:
Why inject providers here instead of just injecting the thing itself?

Reply:
😮 Didn't know I could do that. I was thinking value objects were different from services somehow.